src/SINDYPI_fit.jl

```diff
@@ -0,0 +1,457 @@
+  for i in 1:nvar
+      push!(table, s[i])
+  end
+end
+
+  if npoly >= 2
+      for i in 1:nvar
+          for j in i:nvar
+              push!(table, string(s[i], s[j]))
+          end
+      end
+  end
+
+
+  if npoly >= 3
+      for i in 1:nvar
+          for j in i:nvar
+              for k in j:nvar
+                push!(table, string(s[i], s[j], s[k]))
+          end
+          end
+      end
+  end
+
+
+  if npoly >= 4
+      for i in 1:nvar
+          for j in i:nvar
+              for k in j:nvar
+                  for l in k:nvar
+                    push!(table, string(s[i], s[j], s[k], s[l]))
+
+          end
+          end
+      end
+  end
+end
+
+if npoly >= 5
+  for i in 1:nvar
+      for j in i:nvar
+          for k in j:nvar
+              for l in k:nvar
+                  for m in l:nvar
+                push!(table, string(s[i], s[j], s[k], s[l], s[m]))
+
+      end
+      end
+  end
+end
+end
+end
+
+if npoly >= 6
+  for i in 1:nvar
+      for j in i:nvar
+          for k in j:nvar
+              for l in k:nvar
+                  for m in l:nvar
+                      for n in m:nvar
+                push!(table, string(s[i], s[j], s[k], s[l], s[m], s[n]))
+
+      end
+      end
+  end
+end
+end
+end
+end
+
+if ntrig >=1
+ for i in 1:nvar
+     push!(table, string("sin", "", s[i]))
+ end
+
+ for i in 1:nvar
+     push!(table, string("cos", "", s[i]))
+ end
+end
+
+if ntrig >=2
+ for i in 1:nvar
+     push!(table, string("cos^{2}", "", s[i]))
+ end
+end
+
+if ntrig >=3
+  for i in 1:nvar
+      for j in 1:nvar
+          push!(table, string(s[i], "*", s[i], "sin", "", s[j], "cos", "", s[j]))
+      end
+  end
+end
+
+if ntrig >=4
+  for i in 1:nvar
+      for j in 1:nvar
+          push!(table, string(s[i], "*", s[i], "sin", "", s[j]))
+      end
+  end
+
+  for i in 1:nvar
+      for j in 1:nvar
+          push!(table, string(s[i], "*", s[i], "cos", "", s[j]))
+      end
+  end
+
+  for i in 1:nvar
+      for j in 1:nvar
+          push!(table, string("sin", "", s[i], "cos", "", s[j]))
+      end
+  end
+end
+
+
+tablecopy = deepcopy(table)
+
+for i in 1:length(table)-1
+  push!(table, string("dot", "(", s[w], ")" , "", tablecopy[i+1]))
+end
+
+
+String_table = hcat(table, Xi[w])
+
+Store_Result[w] = String_table
+
+Index_nonzero[w] = findall(x->x !=0, Store_Result[w][:,2])
+
+ODE_Index = Index_nonzero[w]
+
+ODE_Strings = Store_Result[w][:,1][ODE_Index]
+
+ODE_values = Xi[w][ODE_Index]
+
+ODE_Print = []
+push!(ODE_Print, "dot", s[w])
+push!(ODE_Print, "", "=")
+
+
+global count = 0
+for i = 1:length(ODE_Strings)
+    global count
+ if count >0
+    if string(round(ODE_values[i]))[1] == '-'
+        pop!(ODE_Print)
+    end
+  end
+
+    push!(ODE_Print, string(string(round(ODE_values[i], digits = 6)), ODE_Strings[i]))
+    push!(ODE_Print, "", "+")
+
+    count += 1
+end
+
+pop!(ODE_Print)
+
+ODE_Print = join(ODE_Print)
+
+println("The differential equation for ", s[w], " is"  )
+
+println(ODE_Print)
+end

```

Comment:
If you use VSCode or atom, you could get a formatting of the code, which is really helpful for debugging.